### PR TITLE
gas consumption vs gas provided

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
use the actual gas consumed for txs preselection
don't limit intrashard txs due to stuck status

### CODE REVIEW

**Review by iulianpascalau — DISMISSED** [review]
checked with the first implementation, looks good :+1:

**Review by iulianpascalau — DISMISSED** [review]

File: process/block/shardblock.go [hunk 1]
```diff
@@ -596,7 +596,7 @@ func (sp *shardProcessor) indexBlockIfNeeded(
 		return
 	}
 
-	gasConsumedInHeader := sp.baseProcessor.gasConsumedProvider.TotalGasConsumed()
+	gasConsumedInHeader := sp.baseProcessor.gasConsumedProvider.TotalGasProvided()
```
Comment (SebastianMarian) [inline process/block/shardblock.go:599]: gasProvidedInHeader, and maybe we should change also in HeaderGasConsumption DTO (L610) from GasConsumed to GasProvided

**Review by gabi-vuls — DISMISSED** [review]
System test passed. 
Errors are safe and will be solved soon. 
image

**Review by iulianpascalau — APPROVED** [review]

File: process/block/metablock.go [hunk 2]
```diff
@@ -621,7 +621,7 @@ func (mp *metaProcessor) indexBlock(
 		Header:         metaBlock,
 		SignersIndexes: signersIndexes,
 		HeaderGasConsumption: indexer.HeaderGasConsumption{
-			GasConsumed:    gasConsumedInHeader,
+			GasProvided:    gasConsumedInHeader,
```
Comment (SebastianMarian) [inline process/block/metablock.go:624]: gasProvidedInHeader

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
Errors are safe and will be solved with PR #[3623](https://github.com/ElrondNetwork/elrond-go/pull/3623)
image

File: process/block/preprocess/gasComputation.go [hunk 3]
```diff
@@ -124,17 +124,17 @@ func (gc *gasComputation) GasPenalized(hash []byte) uint64 {
...
 }
 
-// TotalGasProvided gets the total gas consumed
+// TotalGasProvided gets the total gas provided
 func (gc *gasComputation) TotalGasProvided() uint64 {
-	totalGasConsumed := uint64(0)
+	totalGasProvided := uint64(0)
 
 	gc.mutGasConsumed.RLock()
 	for _, gasConsumed := range gc.gasConsumed {
```
Comment (SebastianMarian) [inline process/block/preprocess/gasComputation.go:132]: gasProvided and gc.gasProvided ?

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**